src/index.ts

```diff
@@ -0,0 +1,460 @@
+		arr = arr.slice();
+		return Array.from({ length: amount }, (): V => arr.splice(Math.floor(Math.random() * arr.length), 1)[0]);
+	}
+
+	/**
+	 * Obtains unique random key(s) from this collection. This relies on {@link Collection#keyArray}, and thus the caching
+	 * mechanism applies here as well.
+	 * @param {number} [amount] Amount of keys to obtain randomly
+	 * @returns {*|Array<*>} A single key if no amount is provided or an array
+	 */
+	public randomKey(): K;
+	public randomKey(amount: number): K[];
+	public randomKey(amount?: number): K | K[] {
+		let arr = this.keyArray();
+		if (typeof amount === 'undefined') return arr[Math.floor(Math.random() * arr.length)];
+		if (arr.length === 0 || !amount) return [];
+		const rand: K[] = Array.from({ length: amount });
+		arr = arr.slice();
+		for (let i = 0; i < amount; i++) rand[i] = arr.splice(Math.floor(Math.random() * arr.length), 1)[0];
+		return rand;
+	}
+
+	/**
+	 * Searches for a single item where the given function returns a truthy value. This behaves like
+	 * [Array.find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find).
+	 * <warn>All collections used in Discord.js are mapped using their `id` property, and if you want to find by id you
+	 * should use the `get` method. See
+	 * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get) for details.</warn>
+	 * @param {Function} fn The function to test with (should return boolean)
+	 * @param {*} [thisArg] Value to use as `this` when executing function
+	 * @returns {*}
+	 * @example collection.find(user => user.username === 'Bob');
+	 */
+	public find(fn: (value: V, key: K, collection: this) => boolean, thisArg?: any): V | undefined {
+		if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
+		for (const [key, val] of this) {
+			if (fn(val, key, this)) return val;
+		}
+		return undefined;
+	}
+
+	/**
+	 * Searches for the key of a single item where the given function returns a truthy value. This behaves like
+	 * [Array.findIndex()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex),
+	 * but returns the key rather than the positional index.
+	 * @param {Function} fn The function to test with (should return boolean)
+	 * @param {*} [thisArg] Value to use as `this` when executing function
+	 * @returns {*}
+	 * @example collection.findKey(user => user.username === 'Bob');
+	 */
+	public findKey(fn: (value: V, key: K, collection: this) => boolean, thisArg?: any): K | undefined {
+		if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
+		for (const [key, val] of this) {
+			if (fn(val, key, this)) return key;
+		}
+		return undefined;
+	}
+
+	/**
+	 * Removes entries that satisfy the provided filter function.
+	 * @param {Function} fn Function used to test (should return a boolean)
+	 * @param {Object} [thisArg] Value to use as `this` when executing function
```

Comment:
Shouldn't this be {*} instead of {Object} to keep consistency with the rest of the methods?

Reply:
Would it not be better to type the `this` parameter of the callback functions with that?